gitpod-io/gitpod
# [Stripe] Enable automatic tax on transactions

## Description


Enable automatic tax on Stripe transactions:
- [x] Since we're already asking users for a country when they provide a payment method, we might as set their customer's country by default
- [x] Refactor StripeService methods to utilize `attributionId` more and pass around IDs instead of (possibly outdated) Stripe objects
- [x] When creating new subscriptions in supported Stripe Tax regions, pass along `automatic_tax: { enabled: true }` to enable automatic tax collection for these subscriptions

## Related Issue(s)

Fixes https://github.com/gitpod-io/gitpod/issues/13000

## How to test


1. Create a team called "Gitpod [Something]"
2. Add a payment method from a tax-supported country (i.e. any country from the EU or the UK)
3. Once subscribed, open your Stripe customer portal: the country in your billing address should be the same as in 2. (and not "United States" by default)
4. In the Stripe admin dashboard, automatic tax collection should be enabled for your subscription
5. Now cancel the subscription (or create a new team -- should work the same)
6. Add a payment method from a tax-unsupported country (e.g. Canada)
7. Subscribing should still work successfully
8. In the Stripe admin dashboard, automatic tax collection should **not** be enabled for your new subscription

## Release Notes

```release-note
NONE
```

## Documentation


## Werft options:
